docs/en/advanced-concepts/android-testing-with-docker.md

```diff
@@ -0,0 +1,66 @@
+## Android Tests With Docker
+
+When time comes to run appium tests on your own CI (like Jenkins) you would probably think about using docker.
+In order to run Android tests you will also need to have an emulator. Fortunately you can find official docker image that has both Appium and Android Emulator bundled inside: 
+https://hub.docker.com/r/appium/appium-emulator/
+ 
+All you need to know is just 2 simple tricks:
+1. Run ADB server using `-a` switch in the docker container running emulator. ADB will listen to all connected network interfaces, so you can connect emulator from another running container
+2. You will need to setup port forwarding from the container running Appium to the container running emulator. 
+Appium uses a `bootstrap-port` to setup communication to the emulator and it will bind to the localhost by default.
+But since emulator running in another container, you will need to bind that local `bootstrap-port` to the same port on the emulator container
+
+Here is a sample docker compose file that does all these tricks:
+```yaml
+version: "2"
+
+services:
+  emulator:
+    container_name: emulator
+    image: appium/appium-emulator
+    command: >
+          bash -c "adb -a nodaemon server &
+                   emulator64-arm -avd Nexus -no-boot-anim -no-window -noaudio -gpu off"
+
+  appium:
+    container_name: appium
+    image: appium/appium-emulator
```

Comment:
@SrinivasanTarget the idea behind running emulator and appium separately is very simple. Emulator takes a lot of time to start and become ready when running in docker (tens of minutes on our CI), so it make sense to start this container only once and then connect to it on each CI run. Also I found that appium server can often became non-responsive if for some reason I decide to cancel running rspec test session and start a new one. I have to restart appium server quite often, so I would not like to run it as a daemon service in docker. Unlike emulator it is better to start this service on each CI test run.
As for the docker image - I took it here just for a sample, since it contains both appium and emulator. Speaking honestly, I run a custom image on my own. So we can change that part of a doc if you like